Updated all acceptance tests but one to start strong

lucatume opened the pull request:
Ticket: https://central.tri.be/issues/42077

This PR fixes the currently failing or broken existing TEC acceptance tests. Of note:

* division of testing phases: all the modified tests use the *arrange, act, assert* structure to avoid assertions where those do not belong
* much set up is done using WordPress methods using the WPBootrstrapper module
* the tests should now be agnostic of the initial db configuration

borkweb commented on tests/acceptance/CreateATagCept.php:
```diff
@@ -1,17 +1,32 @@
+$I = new AcceptanceTester($scenario);
+
+$I->am('administrator');
+$I->wantTo("verify that a category can be created");
```
Copy/paste error here, I think. Should be `verify that a tag can be created`

borkweb commented:
:+1:

lucatume commented:
I will not merge and delete straight to have @rmacdon465 review it too and have a place to discuss changes

lucatume commented:
putting this back into code review as there have been some changes worth noting:

* removed the `phantomjs` suite as we are using webdrivers to do acceptance already
* refactored the CSV import test
